src/modules/site-v2/base/views/auth/oauth.py

```diff
@@ -56,6 +60,29 @@ def authorized(google_bp, token):
   assert user_info.ok
   user_info = {'google': user_info.json()}
   user = create_or_update_google_user(user_info)
-
+  resp = make_response(assign_access_refresh_tokens(user.name, user.roles, session.get("login_referrer")))
+  new_resp = transfer_cart(resp, user)
   flash("Successfully logged in!", 'success')
-  return assign_access_refresh_tokens(user.name, user.roles, session.get("login_referrer"))
+  return new_resp
+
+
+def transfer_cart(resp, user):
+  # checks if a user has a local cart
+  cartID = request.cookies.get('cartID')
+  if cartID:
+    local_cart = Cart(cartID)
+    if len(local_cart['items']):
```

Comment:
@orzut this is a minor stylistic note, but we prefer code-guards that return early (and keep indentation minimal). 
For instance, when there are zero items, immediately delete the cookie and return. That way the rest of the flow can remain at the function level. Example:
``` 
if len(local_cart['items']) == 0:
  res.delete_cookie('cartID')
  return 

# continue code at this level of indentation.
users_cart = Cart.lookup_by_user(user['email'])
# (...)
```